--- a/luci-app-advancedplus/root/etc/init.d/advancedplus
+++ b/luci-app-advancedplus/root/etc/init.d/advancedplus
@@ -137,24 +137,38 @@ else
 fi
 }
 sethttps() {
-    uhttp=`uci -q get uhttpd.main.listen_https | awk -F ' ' '{printf $1}'`
-    if [ "x$(uci -q get advancedplus.@basic[0].uhttps)" = "x1" ] ; then
-	     [  $uhttp ] || {
-	         sed -i "/listen_https/d" /etc/config/uhttpd
-	         /etc/init.d/uhttpd stop
-	         uci -q add_list uhttpd.main.listen_https='0.0.0.0:443'
-	         uci -q add_list uhttpd.main.listen_https='[::]:443'
-	         uci commit uhttpd
-	         /etc/init.d/uhttpd restart
-	    }
+	if [ -n "$(command -v nginx)" ]; then
+		if [ -z "`uci -q get nginx._redirect2ssl`" ]; then
+			uci -q set nginx._redirect2ssl=server
+			uci -q set nginx._redirect2ssl.server_name='_redirect2ssl'
+			uci -q set nginx._redirect2ssl.return='302 https://$host$request_uri'
+			uci -q set nginx._redirect2ssl.access_log='off; # logd openwrt'
+		fi
+    	if [ "x$(uci -q get advancedplus.@basic[0].uhttps)" = "x1" ] ; then
+    		uci -q del nginx.default_server.listen
+    		uci -q add_list nginx.default_server.listen='80'
+    		uci -q add_list nginx.default_server.listen='[::]:80'
+    		uci -q del nginx._redirect2ssl.listen
+    		uci -q add_list nginx._redirect2ssl.listen='80 default_server'
+    		uci -q add_list nginx._redirect2ssl.listen='[::]:80 default_server'
+    	else
+	         uci -q del nginx._redirect2ssl.listen
+	         uci -q add_list nginx._redirect2ssl.listen='80'
+	         uci -q add_list nginx._redirect2ssl.listen='[::]:80'
+	         uci -q del nginx.default_server.listen
+	         uci -q add_list nginx.default_server.listen='80 default_server'
+	         uci -q add_list nginx.default_server.listen='[::]:80 default_server'
+    	fi
+		uci commit nginx
+    	/etc/init.d/nginx reload
     else
-        [  $uhttp ] && 
-	{  
-	     /etc/init.d/uhttpd stop
-	    sed -i "/listen_https/d" /etc/config/uhttpd
-	    uci commit uhttpd
-	    /etc/init.d/uhttpd restart
-	}
+        if [ "x$(uci -q get advancedplus.@basic[0].uhttps)" = "x1" ] ; then
+	         uci -q set uhttpd.main.redirect_https='1'
+		else
+	         uci -q set uhttpd.main.redirect_https='0'
+        fi
+		uci commit uhttpd
+	    /etc/init.d/uhttpd reload
     fi
 }
 
@@ -196,24 +210,21 @@ set_ttyd() {
 }
 
 dhcp_domain() {
+    domain_name="time.android.com"
+    existing_records="$(uci -q get dhcp.$domain_name)"
     if [ "x$(uci -q get advancedplus.@basic[0].dhcp_domain)" = "x1" ] ; then
-        local domain_name="time.android.com"
         local domain_ip="203.107.6.88"
-        existing_records=$(uci show dhcp | grep "dhcp.@domain\[[0-9]\+\].name='$domain_name'")
         if [ -z "$existing_records" ]; then
-        uci add dhcp domain
-        uci -q set "dhcp.@domain[-1].name=$domain_name"
-        uci -q set "dhcp.@domain[-1].ip=$domain_ip"
-        uci commit dhcp
+        uci set dhcp.$domain_name=domain
+        uci -q set dhcp.$domain_name.name="$domain_name"
+        uci -q set dhcp.$domain_name.ip="$domain_ip"
         fi
     else
-       existing_records=$(uci show dhcp | grep "dhcp.@domain\[[0-9]\+\].name='$domain_name'")
-       if [ -z "$existing_records" ]; then
-       uci delete dhcp.@domain[-1].name
-       uci delete dhcp.@domain[-1].ip
-       uci commit dhcp
+       if [ -n "$existing_records" ]; then
+       uci delete dhcp.$domain_name
        fi
     fi
+    uci commit dhcp
 }
 advancedset(){
 dev=`ifconfig | grep "Point-to-Point" | cut -d " " -f1`
@@ -233,8 +244,10 @@ tsoset
     fi
     #cpumode=`uci -q get advancedplus.@basic[0].cpumode`
     #[ ! $cpumode ] || cpumodeset $cpumode   /usr/bin/zsh
+    if [ "$(which zsh)" ]; then
     sed -i "\/bin\/zsh/d" /etc/profile
     [ "x$(uci -q get advancedplus.@basic[0].usshmenu)" = "x1" ] || echo '/usr/bin/zsh'  >> /etc/profile
+    fi
     uci commit netwizard 
 }
 

--- a/luci-app-advancedplus/luasrc/controller/advancedplus.lua
+++ b/luci-app-advancedplus/luasrc/controller/advancedplus.lua
@@ -21,9 +21,9 @@ function index()
 	
 	entry({"admin","system","advancedplus","advancededit"},cbi("advancedplus/advancededit"),_("Advanced Edit"),10).leaf = true
 	entry({"admin","system","advancedplus","advancedset"},cbi("advancedplus/advancedset"),_("Advanced Setting"),20).leaf = true
-	entry({"admin","system","advancedplus","advancedipk"},cbi("advancedplus/advancedipk", {hideapplybtn=true, hidesavebtn=true, hideresetbtn=true}),_("Loading plugins"),30).leaf = true
 	if nixio.fs.access('/www/luci-static/kucat/css/style.css') then
 	    entry({"admin","system","advancedplus","kucatset"},cbi("advancedplus/kucatset"),_("KuCat Theme Config"),40).leaf = true
+	    entry({"admin", "system","advancedplus","kucatupload"}, form("advancedplus/kucatupload"), _("Desktop background upload"), 80).leaf = true
 	end
 	if nixio.fs.access('/www/luci-static/argon/css/cascade.css') then
 	    entry({"admin", "system", "advancedplus","argon-config"}, form("advancedplus/argon-config"), _("Argon Config"), 50).leaf = true
@@ -32,7 +32,6 @@ function index()
 	    entry({"admin", "system",  "advancedplus","design-config"}, form("advancedplus/design-config"), _("Design Config"), 60).leaf = true
 	end
 	entry({"admin", "system","advancedplus","upload"}, form("advancedplus/upload"), _("Login Background Upload"), 70).leaf = true
-	entry({"admin", "system","advancedplus","kucatupload"}, form("advancedplus/kucatupload"), _("Desktop background upload"), 80).leaf = true
 	
 
 	entry({"admin", "system", "advancedplus", "advancedrun"}, call("advanced_run"))

--- a/luci-app-advancedplus/luasrc/model/cbi/advancedplus/advancedset.lua
+++ b/luci-app-advancedplus/luasrc/model/cbi/advancedplus/advancedset.lua
@@ -24,11 +24,9 @@ ul:depends("qos", true)
 
 e = t:option(Flag, "uhttps",translate('Accessing using HTTPS'), translate('Open the address in the background and use HTTPS for secure access'))
 
+if fs.access('/usr/bin/zsh') then
 e = t:option(Flag, "usshmenu",translate('No backend menu required'), translate('OPENWRT backend and SSH login do not display shortcut menus'))
-
-e = t:option(Flag, "wizard",translate('Hide Wizard'), translate('Show or hide the setup wizard menu'))
-e.default = "0"
-e.rmempty = false
+end
 
 e = t:option(Flag, "tsoset",translate('TSO optimization for network card interruption'), translate('Turn off the TSO parameters of the INTEL225 network card to improve network interruption'))
 e.default = "1"
