--- a/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js
+++ b/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js
@@ -69,6 +69,15 @@ return view.extend({
 			p[i].value('ACCEPT', _('accept'));
 		}
 
+		if (L.hasSystemFeature('fullconenat')) {
+		o = s.option(form.ListValue, 'fullcone', _('Enable FullCone NAT'), _('Using FullCone NAT can improve gaming performance effectively'));
+			o.value("0", _("Disable"))
+			o.value("1", _("Compatible Mode"))
+			o.value("2", _("High Performing Mode"))
+		if (fw4)
+			o = s.option(form.Flag, 'fullcone6', _('Enable FullCone NAT6'));
+		}
+
 		/* Netfilter flow offload support */
 
 		if (L.hasSystemFeature('offloading')) {

--- a/luci-app-firewall/po/zh_Hans/firewall.po
+++ b/luci-app-firewall/po/zh_Hans/firewall.po
@@ -1520,3 +1520,16 @@ msgstr "最多 65536 个条目。"
 #: applications/luci-app-firewall/htdocs/luci-static/resources/tools/firewall.js:339
 msgid "valid firewall mark"
 msgstr "有效的防火墙标记"
+
+msgid "Enable FullCone NAT"
+msgstr "启用 FullCone NAT"
+
+msgid "Enable FullCone NAT6"
+msgstr "启用 FullCone NAT6"
+
+msgid "Compatible Mode"
+msgstr "兼容模式"
+
+msgid "High Performing Mode"
+msgstr "高性能模式"
+