--- a/luci-app-nekoclash/root/etc/uci-defaults/99_neko
+++ b/luci-app-nekoclash/root/etc/uci-defaults/99_neko
@@ -24,11 +24,31 @@ fi
 echo "[ `date +%T` ] - Checking Files"
 
 files_check() {
-    uci set uhttpd.main.index_page='index.php'
-    uci set uhttpd.main.interpreter='.php=/usr/bin/php-cgi'
-    uci commit uhttpd
+    if [ -n "$(command -v nginx)" ]; then
+        cat << EOF > /etc/nginx/nekoclash.locations
+location /nekoclash {
+    alias /www/nekoclash;
+    index index.php index.html index.htm;
+
+    location ~ \.php$ {
+        if (!-f \$request_filename) {
+            return 404;
+        }
+        fastcgi_pass unix:/var/run/php8-fpm.sock;
+        fastcgi_index index.php;
+        include fastcgi_params;
+        fastcgi_param SCRIPT_FILENAME \$request_filename;
+    }
+}
+EOF
+        /etc/init.d/nginx restart
+    else
+        uci set uhttpd.main.index_page='index.php'
+        uci set uhttpd.main.interpreter='.php=/usr/bin/php-cgi'
+        uci commit uhttpd
+
+        /etc/init.d/uhttpd restart
+    fi
 
     case "$arch" in
     aarch64)
--- a/luci-app-nekoclash/Makefile
+++ b/luci-app-nekoclash/Makefile
@@ -15,7 +15,7 @@ define Package/$(PKG_NAME)
 	SUBMENU:=3. Applications
 	TITLE:=LuCI support for mihomo
 	PKGARCH:=all
-	DEPENDS:=+kmod-tun +bash +curl +jq +php8 +php8-cgi +php8-mod-curl +firewall4  +nftables
+	DEPENDS:=+kmod-tun +bash +curl +jq +php8 +php8-fpm +php8-cgi +php8-mod-curl +firewall4 +nftables
 endef
 
 define Package/$(PKG_NAME)/description
--- a/luci-app-nekoclash/htdocs/nekoclash/devinfo.php
+++ b/luci-app-nekoclash/htdocs/nekoclash/devinfo.php
@@ -213,25 +213,6 @@
     </style>
 </head>
 <body>
-<div id="player" onclick="toggleAnimation()"> 
-    <p id="hidePlayer">Mihomo</p>
-    <p id="timeDisplay">00:00</p>
-    <audio id="audioPlayer" controls>  
-        <source src="" type="audio/mpeg">
-        您的浏览器不支持音频播放。
-    </audio>
-    <br>
-    <div id="controls">
-        <button id="prev" class="rounded-button">⏮️</button>
-        <button id="orderLoop" class="rounded-button">🔁</button>
-        <button id="play" class="rounded-button">⏸️</button>
-        <button id="next" class="rounded-button">⏭️</button> 
-    </div>  
-</div>
-<div id="mobile-controls">
-    <button id="togglePlay" class="rounded-button">播放/暂停</button>
-    <button id="toggleEnable" class="rounded-button">启用/禁用</button>
-</div>
 <div id="tooltip"></div>
 
 <script>
