--- a/wifi-scripts/files/etc/hotplug.d/ieee80211/10-wifi-detect
+++ b/wifi-scripts/files/etc/hotplug.d/ieee80211/10-wifi-detect
@@ -1,5 +1,10 @@
 #!/bin/sh
 
 [ "${ACTION}" = "add" ] && {
-	/sbin/wifi config
+	for i in $(seq 1 3); do
+		/sbin/wifi config
+		grep -q '"wlan"' /etc/board.json && break
+		sleep 1s
+	done
+	[ -z "$(uci -q get base_config.@status[0].wifi)" ] && . /sbin/mywifi
 }

--- a/wifi-scripts/files/lib/wifi/mac80211.uc
+++ b/wifi-scripts/files/lib/wifi/mac80211.uc
@@ -77,7 +77,6 @@ set ${s}.${id}
 set ${s}.band='${lc(band_name)}'
 set ${s}.channel='${channel}'
 set ${s}.htmode='${htmode}'
-set ${s}.disabled='1'
 
 set ${si}=wifi-iface
 set ${si}.device='${name}'